src/account_manager/operations/address_generation.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::atomic::Ordering;
+
+use iota_client::{
+    block::address::Address,
+    secret::{GenerateAddressOptions, SecretManage, SecretManager},
+};
+
+use crate::account_manager::AccountManager;
+#[cfg(all(feature = "events", feature = "ledger_nano"))]
+use crate::events::types::{AddressData, WalletEvent};
+
+impl AccountManager {
+    /// Generate an address without storing it
+    /// ```ignore
+    /// let public_addresses = account_manager
+    ///     .generate_address(
+    ///         0,
+    ///         false,
+    ///         0,
+    ///         false,
+    ///     )
+    ///     .await?;
+    /// ```
+    pub async fn generate_address(
+        &self,
+        account_index: u32,
+        internal: bool,
+        address_index: u32,
+        options: Option<GenerateAddressOptions>,
+    ) -> crate::Result<Address> {
+        let address = match &*self.secret_manager.read().await {
+            #[cfg(feature = "ledger_nano")]
+            SecretManager::LedgerNano(ledger_nano) => {
+                // If we don't sync, then we want to display the prompt on the ledger with the address. But the user
+                // needs to have it visible on the computer first, so we need to generate it without the
+                // prompt first
+                if options.clone().unwrap_or_default().ledger_nano_prompt {
```

Comment:
`.as_ref().map_or()` if ever you want to avoid the clone + newly created object just to get a field